api/src/main/java/io/minio/CopyConditions.java

```diff
@@ -89,6 +89,20 @@ public void setMatchETagNone(String etag) throws InvalidArgumentException {
   }
 
   /**
+   * Set metadata directive to specify whether the metadata
+   * is copied from the source object or replaced with metadata
+   * provided in the request.
+   */
+  public void setMetadataDirective(String value) {
+    if (value == null) {
+      // http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectCOPY.html
+      // Default value is - COPY.
+      value = "COPY";
```

Comment:
The reason i didn't add enum is because it wasn't needed,  server returns an error anyways if the wrong metadata directive was set.  Doing an enum disadvantage is that we are defining new types where none exists from a server perspective. 

Users of aws-sdk would have a hard time looking up these enums. Perhaps a strings constant is better..